lib/sles4sap/sap_deployment_automation_framework/deployment.pm

```diff
@@ -1102,4 +1104,73 @@ sub sdaf_deployment_reused {
     return $deployment_id;
 }
 
+=head2 validate_components
+
+    validate_components(components=>['db_install', 'db_ha']);
+
+Checks if components list is valid and supported by code. Croaks if not.
+Currently supported components are:
+
+=over
+
+=item * B<components>: B<ARRAYREF> of components that should be installed.
+    Supported values:
+        db_install : Basic DB installation
+        db_ha : Database HA setup
+        nw_pas : Installs primary application server (PAS)
+        nw_aas : Installs additional application server (AAS)
+        nw_ensa : Installs enqueue replication server (ERS)
+
+=back
+
+=cut
+
+sub validate_components {
+    my (%args) = @_;
+    croak '$args{components} must be an ARRAYREF' unless ref($args{components}) eq 'ARRAY';
+
+    my %valid_components = ('db_install' => 'Basic DB installation.',
+        db_ha => 'db_ha : Database HA setup',
+        nw_pas => 'db_pas : Installs primary application server (PAS)',
+        nw_aas => 'nw_aas : Installs additional application server (AAS)',
+        nw_ensa => 'nw_ensa : Installs enqueue replication server (ERS)');
+
+    for my $component (@{$args{components}}) {
+        croak "Unsupported component: '$component'\nSupported values:\n" . join("\n", values(%valid_components))
+          unless grep /^$component$/, keys(%valid_components);
+    }
+    # need to return positive value for unit test to work properly
+    return 1;
+}
+
+=head2 get_fencing_mechanism
+
+    get_fencing_mechanism(components=>['db_install', 'db_ha']);
+
+Converts fencing type naming used by existing OpenQA tests into corresponding value accepted by SDAF setting (tfvars).
+Value is retrieved as a mandatory OpenQA setting 'SDAF_FENCING_MECHANISM'.
+
+B<Value conversion:>
+
+=over
+
+=item * B<msi> =>  'AFA' - Azure fencing agent
+
+=item * B<sbd> => 'ISCSI' - ISCSI based SBD fencing device
+
+=item * B<asd> => 'ASD' - Azure shared disk based SBD device
+
+=back
+
+=cut
+
+sub get_fencing_mechanism {
+    # Fencing type must be set as mandatory OpenQA setting to keep value consistent across whole code
+    my $fencing_type = get_required_var('SDAF_FENCING_MECHANISM');
+    my %supported_fencing_values = (msi => 'AFA', sbd => 'ISCSI', asd => 'ASD');
+    die "Fencing type '$fencing_type' is not supported" unless grep /^$fencing_type$/, keys(%supported_fencing_values);
+
```

Comment:
This empty line looks all weird here. I'd remove it.

Reply:
Haha. Fixed.